pandas/tests/io/parser/test_python_parser_only.py

```diff
@@ -314,3 +316,15 @@ def test_malformed_skipfooter(python_parser_only):
     msg = "Expected 3 fields in line 4, saw 5"
     with pytest.raises(ParserError, match=msg):
         parser.read_csv(StringIO(data), header=1, comment="#", skipfooter=1)
+
+
+def test_file_descriptor_leak(python_parser_only):
+    # GH 31488
+    parser = python_parser_only
+    with open("empty.csv", "w"):
+        pass
+
+    proc = psutil.Process()
+    with pytest.raises(EmptyDataError):
+        parser.read_csv("empty.csv")
+    assert not proc.open_files()
```

Comment:
I would check that the list of open files does not contain our file, not that it's empty (more vulnerable to failure if the testing environment happen to have other files open).

Reply:
I have now changed the test to a before/after check that triggers when the results differ.